src/system/core.cpp

```diff
@@ -1,2 +1,9 @@
 #include "core.h"
 
+#include "connectivity/wifi.h"
+
+
+IrrigationSystem::IrrigationSystem () {
+    WifiInitialize ( );
```

Comment:
I think that shouldn't be in the constructor, the IrrigationSystem class should add a `init` function which will call the WiFiInitialize.

The main reason is that the constructors are designed to build or initialize a class, in this case, we are not just configuring the wifi, we are initializing it and connecting to wifi, sending logs etc.